File: src/main/java/info/skyblond/archivedag/config/JwtProperties.java
package info.skyblond.archivedag.config;

import io.jsonwebtoken.SignatureAlgorithm;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.concurrent.TimeUnit;

@Getter
@Setter
@ConfigurationProperties(value = "jwt")
public class JwtProperties {
    private String secret = null;
    private long expireInDuration = 5;
    private TimeUnit expireInUnit = TimeUnit.MINUTES;
    private SignatureAlgorithm signatureAlgorithm = SignatureAlgorithm.HS512;
}

File: backend/src/main/kotlin/info/skyblond/archivedag/arudaz/config/JwtProperties.kt
package info.skyblond.archivedag.arudaz.config

import io.jsonwebtoken.SignatureAlgorithm
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import java.util.concurrent.TimeUnit

@ConstructorBinding
@ConfigurationProperties(value = "jwt")
data class JwtProperties(
    val secret: String? = null,
    val expireInDuration: Long = 5,
    val expireInUnit: TimeUnit = TimeUnit.MINUTES,
    val signatureAlgorithm: SignatureAlgorithm = SignatureAlgorithm.HS512
)
